zaza/openstack/charm_tests/vault/setup.py

```diff
@@ -187,12 +188,13 @@ def auto_initialize(cacert=None, validation_application='keystone', wait=True,
         validate_ca(cacertificate, application=validation_application)
         # Once validation has completed restart nova-compute to work around
         # bug #1826382
+        nova_compute_app_name = get_app_names_for_charm('nova-compute')[0]
         cmd_map = {
             'nova-cloud-controller': ('systemctl restart '
                                       'nova-scheduler nova-conductor'),
-            'nova-compute': 'systemctl restart nova-compute',
+            nova_compute_app_name: 'systemctl restart nova-compute',
         }
-        for app in ('nova-compute', 'nova-cloud-controller',):
+        for app in (nova_compute_app_name, 'nova-cloud-controller',):
```

Comment:
I might be misunderstanding the intent here, but if you anticipate that there
might be several apps that use the `nova-compute` charm, wouldn't you want to
restart services on all of those? However, you only store the first application
in line 191.

Reply:
yes, that was partial future-proofing code, it is easy to make it fully future proof, making those changes now. I was initially just fixing the error where it was hardcoded.